### DISCUSSION TITLE
Custom Component differentiating inline Code vs block-level Code

### DISCUSSION THREAD

**Discussion by xHomu** [mdx-js/mdx#2108] [Q&A] [answered]
My current custom component for `components = { code: (props: any) => SyntaxHighlightCode(props)}` couldn't differentiate between inline vs block level code.

What's the best way to differentiate between `inline-code` vs

```
block level code?
```

Which has the undesireable effect of breaking up inline code from its paragraph. 

I suspect the answer is leave `<code>` alone and insert a custom component for `<pre>` instead?

**Comment by wooorm** [comment, marked as answer]
Depends on what you want, but generally I’d say, look at props. If you want to highlight code there is a class specifying that language.
Your last paragraph might work too, yeah.

**Reply by prmichaelsen** [reply]
For posterity, the last paragraph does work.
